addons/source-python/packages/source-python/translations/strings.py

```diff
@@ -250,11 +244,29 @@ def get_string(self, language=None, **tokens):
             # Possibly raise an error silently here
             return ''
 
-        # Update the stored tokens with the given ones
-        self.tokens.update(tokens)
+        # Expose all TranslationStrings instances in self.tokens
+        exposed_tokens = {}
+        for token_name, token in self.tokens.items():
+            if isinstance(token, TranslationStrings):
+
+                # Pass additional tokens - these will be used to format
+                # the string
+                token = token.get_string(language, **tokens)
+
+            exposed_tokens[token_name] = token
+
+        # Expose all TranslationsStrings instances in **tokens
```

Comment:
There is no need to do that, but it removes duplicated code. And in my opinion it's easier to read the code, because it's easier to see the difference between both calls.